classport-agent/src/test/java/ClassportAgentTest.java

```diff
@@ -0,0 +1,164 @@
+
+
+    private byte[] loadClassFromFile(File classFile) throws IOException {
+    try (InputStream inputStream = new FileInputStream(classFile)) {
+        return inputStream.readAllBytes();
+        }
+    }
+
+    private boolean compareFiles(File file1, File file2) throws IOException {
+        return Arrays.equals(Files.readAllBytes(file1.toPath()), Files.readAllBytes(file2.toPath()));
+    }
+
+    
+
+    private HashMap<String, ClassportInfo> getSBOM() {
+        HashMap<String, ClassportInfo> sbom = new HashMap<>();
+        sbom.put("com.example:test-agent-app:jar:1.0-SNAPSHOT", 
+                 createClassportInfo("com.example", "1.0-SNAPSHOT", 
+                                     "com.example:test-agent-app:jar:1.0-SNAPSHOT", 
+                                     "test-agent-app", false, 
+                                     "com.example:test-agent-app:jar:1.0-SNAPSHOT", 
+                                     new String[]{"com.google.code.gson:gson:jar:2.11.0", 
+                                                  "org.apache.commons:commons-lang3:jar:3.17.0"}));
+        sbom.put("com.google.code.gson:gson:jar:2.11.0", 
+                 createClassportInfo("com.google.code.gson", "2.11.0", 
+                                     "com.google.code.gson:gson:jar:2.11.0", 
+                                     "gson", true, 
+                                     "com.example:test-agent-app:jar:1.0-SNAPSHOT", 
+                                     new String[]{"com.google.errorprone:error_prone_annotations:jar:2.27.0"}));
+        sbom.put("org.apache.commons:commons-lang3:jar:3.17.0", 
+                 createClassportInfo("org.apache.commons", "3.17.0", 
+                                     "org.apache.commons:commons-lang3:jar:3.17.0", 
+                                     "commons-lang3", true, 
+                                     "com.example:test-agent-app:jar:1.0-SNAPSHOT", 
+                                     new String[]{"org.apache.commons:commons-text:jar:1.12.0"}));
+        return sbom;
+    }
+    
+    private ClassportInfo createClassportInfo(String group, String version, String id, 
+                                          String artefact, boolean isDirectDependency, 
+                                          String sourceProjectId, String[] childIds) {
+        return new ClassportInfo() {
+            @Override public String group() { return group; }
+            @Override public String version() { return version; }
+            @Override public String id() { return id; }
+            @Override public String artefact() { return artefact; }
+            @Override public boolean isDirectDependency() { return isDirectDependency; }
+            @Override public String sourceProjectId() { return sourceProjectId; }
+            @Override public String[] childIds() { return childIds; }
+            @Override public Class<? extends Annotation> annotationType() { return ClassportInfo.class; }
+        };
+    }
```

Comment:
Ideally all of this should be handled by `getSBOM` in the main code. You would need to use that since it takes a `jar` file. You could create one manually with all these dependencies.